### PR TITLE
Code cleanup for qdel

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
I want to put my hands in the engine. If you don't want me touching this I am not hurt if its closed

Added 2 early returns and removes single letter vars
## Why It's Good For The Game
Code improvement
## Changelog
N/A nothing player facing

### CODE REVIEW

File: code/controllers/subsystem/garbage.dm [hunk 1]
```diff
@@ -363,71 +363,78 @@ SUBSYSTEM_DEF(garbage)
 /// Should be treated as a replacement for the 'del' keyword.
 ///
 /// Datums passed to this will be given a chance to clean up references to allow the GC to collect them.
-/proc/qdel(datum/D, force=FALSE, ...)
-	if(!istype(D))
-		del(D)
+/proc/qdel(datum/to_delete, force = FALSE, ...)
```
Comment (LemonInTheDark) [inline code/controllers/subsystem/garbage.dm:366]: this may make mso upset fair warning

Comment (jlsnow301) [reply code/controllers/subsystem/garbage.dm:366]: i'm not opinionated enough on this tbh. 
should i just put back the single letter vars

File: code/controllers/subsystem/garbage.dm [hunk 2]
```diff
@@ -363,71 +363,78 @@ SUBSYSTEM_DEF(garbage)
...
-		if(world.time != start_time)
-			I.slept_destroy++
-		else
-			I.destroy_time += TICK_USAGE_TO_MS(start_tick)
-		if(!D)
+	if(to_delete.gc_destroyed == GC_CURRENTLY_BEING_QDELETED)
+		CRASH("[to_delete.type] destroy proc was called multiple times, likely due to a qdel loop in the Destroy logic")
+
+	if(!isnull(to_delete.gc_destroyed))
+		return
```
Comment (LemonInTheDark) [inline code/controllers/subsystem/garbage.dm:380]:
```suggestion
	if(!isnull(to_delete.gc_destroyed))
		if(to_delete.gc_destroyed == GC_CURRENTLY_BEING_QDELETED)
				CRASH("[to_delete.type] destroy proc was called multiple times, likely due to a qdel loop in the Destroy logic")
		return
```
this is a somewhat hot proc, it is worth being slightly more ugly here

Comment (san7890) [reply code/controllers/subsystem/garbage.dm]: what's the point of having both a CRASH and a return if a CRASH ends the proc

File: code/controllers/subsystem/garbage.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/controllers/subsystem/garbage.dm:366]: the rest of ssgarbage is full of em, but that's just cause mso code 
this is good and fine (I'm mostly just preparing for an i told you so him getting upset isn't the end of the world he just disagrees with us about single letter vars)

File: code/controllers/subsystem/garbage.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/controllers/subsystem/garbage.dm]: because we want to return in all cases

**Review by LemonInTheDark — APPROVED** [review]

File: code/controllers/subsystem/garbage.dm (hunk 2 above)
Comment (san7890) [reply code/controllers/subsystem/garbage.dm]: oh never mind i misread that

File: code/controllers/subsystem/garbage.dm [hunk 3]
```diff
@@ -363,71 +363,77 @@ SUBSYSTEM_DEF(garbage)
 /// Should be treated as a replacement for the 'del' keyword.
 ///
 /// Datums passed to this will be given a chance to clean up references to allow the GC to collect them.
-/proc/qdel(datum/D, force=FALSE, ...)
+/proc/qdel(datum/D, force = FALSE, ...)
```
Comment (LemonInTheDark) [inline code/controllers/subsystem/garbage.dm:366]: I WAS JUST HEDGING MY BETS PUT IT BACK NOOO THE SOULCODE IT BURNS

Comment (jlsnow301) [reply code/controllers/subsystem/garbage.dm:366]: FINE

File: code/controllers/subsystem/garbage.dm [hunk 4]
```diff
@@ -342,92 +342,110 @@ SUBSYSTEM_DEF(garbage)
...
-	var/hard_deletes_over_threshold = 0 //!Number of times hard deletes took longer than the configured threshold
-	var/no_respect_force = 0 //!Number of times it's not respected force=TRUE
-	var/no_hint = 0 //!Number of times it's not even bother to give a qdel hint
-	var/slept_destroy = 0 //!Number of times it's slept in its destroy
-	var/qdel_flags = 0 //!Flags related to this type's trip thru qdel.
-	var/list/extra_details //!Lazylist of string metadata about the deleted objects
+	/// Holds the type as a string for this type
+	var/name = ""
+	/// Total number of times it's passed thru qdel.
+	var/qdels = 0
```
Comment (LemonInTheDark) [inline code/controllers/subsystem/garbage.dm:348]: do you just prefer this or someodd?

Comment (jlsnow301) [reply code/controllers/subsystem/garbage.dm:348]: [AUTODOC](https://github.com/tgstation/tgstation/blob/master/.github/guides/AUTODOC.md#documenting-a-variabledefine)

Comment (LemonInTheDark) [reply code/controllers/subsystem/garbage.dm:348]: https://github.com/SpaceManiac/SpacemanDMM/tree/master/crates/dmdoc#documenting-code

Comment (jlsnow301) [reply code/controllers/subsystem/garbage.dm:348]: I wasn't getting the comment on hover :/

Comment (LemonInTheDark) [reply code/controllers/subsystem/garbage.dm:348]: we should fix that this is accepted syntax

Comment (ZephyrTFA) [reply code/controllers/subsystem/garbage.dm:348]: I'm fine with //! but it does need to be in the readme if we plan on using it

Comment (jlsnow301) [reply code/controllers/subsystem/garbage.dm:348]: //! seems bugged at the moment and does not offer tooltips 
Code_jFoJMJBNXL

Comment (ZephyrTFA) [reply code/controllers/subsystem/garbage.dm:348]: odd, it's supposed to work

Comment (LemonInTheDark) [reply code/controllers/subsystem/garbage.dm:348]: if it's broken we should fix it rather then removing uses of it

Comment (jlsnow301) [reply code/controllers/subsystem/garbage.dm:348]: I'll revert these comments and make an issue report

Comment (LemonInTheDark) [reply code/controllers/subsystem/garbage.dm:348]: o7

**Pull request merged**